pkgs/http/lib/src/client.dart

```diff
@@ -154,7 +154,7 @@ abstract class Client {
 @internal
 Client? get zoneClient {
   final client = Zone.current[#_clientToken];
-  return client == null ? null : (client as Client Function())();
+  return client == null ? null : (client as Client? Function())();
```

Comment:
[optional] If you wanted you could use `?.call` over the `?:`. This could be shortened all the way to a `=>`

```dart
Client? get zoneClient =>
    (Zone.current[#_clientToken] as Client? Function()?)?.call();
```

Reply:
I find the additional question marks harder to read.